src/libasr/asr_verify.cpp

```diff
@@ -441,6 +441,20 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
         verify_unique_dependencies(x.m_dependencies, x.n_dependencies,
                                    x.m_name, x.base.base.loc);
 
+        // Get the x symtab.
+        SymbolTable *x_symtab = x.m_symtab;
+
+        // Dependencies of the function should be from function's parent symbol table.
+        for( size_t i = 0; i < x.n_dependencies; i++ ) {
+            std::string found_dep = x.m_dependencies[i];
+
+            // Get the symbol of the found_dep.
+            ASR::symbol_t* dep_sym = x_symtab->resolve_symbol(found_dep);
+
+            require(dep_sym != nullptr,
+                            "Dependency " + found_dep +  " is inside symbol table " + std::string(x.m_name));
```

Comment:
I think this should be `"Dependency " + found_dep +  " is not* inside symbol table " `

Reply:
correct, "not inside".

Reply:
Yes, you both are right.

I assumed the ``resolve_symbol`` checks in parent scope only and not in the ``current_scope``. Now that i checked it's documentation, it checks both in current & parent scopes. 

The check should will be updated like this :
``SymbolTable *x_parent_symtab = x.m_symtab->parent;`` instead of  ``SymbolTable* x_symtab = x.m_symtab;``
and then "inside" phrase will make sense here.